### PR TITLE
[PPR-341] add cosmosdb account and first db

### PR DESCRIPTION
**Pull request by pasqualespica**
The purpose of this PR is to add cosmos db account and a first db according to [nodo-dei-pagamenti-biz-evt](https://pagopa.atlassian.net/wiki/spaces/PAG/pages/497746877/Design+Review-+Evento+di+pagamento) 
and new logicapp
image

### List of changes

Add : 
  - cosmos db account
  - document db `paymentsdb`
  - new subnet
  - private endpoint

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/dns_private.tf [hunk 1]
```diff
@@ -102,3 +102,21 @@ resource "azurerm_private_dns_zone" "privatelink_azurecr_pagopa" {
 
   tags = var.tags
 }
+
+# MongoDB
+resource "azurerm_private_dns_zone" "privatelink_mongo_cosmos_azure_com" {
```
Comment (infantesimone) [inline src/dns_private.tf:107]: private link for all envs? also for DEV?

File: src/mongodb.tf [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  allowed_virtual_network_subnet_ids = []
+
+  tags = var.tags
+}
+
+# database(s)
+
+# 1) payments db
+resource "azurerm_cosmosdb_mongo_database" "pagopa_payments_cosmosdb_mongodb_db" {
```
Comment (infantesimone) [inline src/mongodb.tf:66]: could it be useful to decouple the cosmos  account definition from the product-specific  database definition?

File: src/dns_private.tf (hunk 1 above)
Comment (pasqualespica) [reply src/dns_private.tf:107]: disable for DEV

File: src/mongodb.tf (hunk 2 above)
Comment (pasqualespica) [reply src/mongodb.tf:66]: do you say to put the db and its collection in the separate TF file?

Comment (infantesimone) [reply src/mongodb.tf:66]: yes, but it's just an idea. We can also refactor in the future (if the number of databases/collections increase).

Comment (pasqualespica) [reply src/mongodb.tf:66]: good tip, i put there into dedicated file

**Pull request merged**